test/e2e/cluster_extension_install_test.go

```diff
@@ -1021,3 +1021,30 @@ func getArtifactsOutput(t *testing.T) {
 		}
 	}
 }
+
+func TestClusterCatalogLabels(t *testing.T) {
```

Comment:
Do we need this separate test in `operator-controller`? It seems like
1. it is functionality that is fully encompassed (and now tested) in catalogd.
2. we are already testing this indirectly via operator-controller's support of catalog selectors in its ClusterExtension API.

Reply:
If catalogd's behavior changes, then this would detect it, but so would other tests. I can live without it

Reply:
> but so would other tests

Right, our other tests depend on that functionality working now.

Reply:
+1 on not testing catalogd's behavior explicitly in operator-controller